Modules/System/Cryptography Management/src/Rfc2898DeriveBytes.Codeunit.al

```diff
@@ -0,0 +1,41 @@
+// ------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+// ------------------------------------------------------------------------------------------------
+
+/// <summary>
+/// Provides helper functions for the Advanced Encryption Standard.
+/// </summary>
+
+codeunit 50101 "Rfc2898DeriveBytes"
+{
+    Access = Public;
+
+    /// <summary>
+    /// Initializes a new instance of the Rfc2898DeriveBytes class providing the Password, Salt, Saltlength and Psuedorandomnumber.
+    /// </summary>
+    /// <param name="Password">Represents the password to be used to initialize a new instance of Rfc2898DeriveBytes</param>
+    /// <param name="Salt">Represents the salt to be used to initialize a new instance of Rfc2898DeriveBytes</param>
+    /// <param name="DesiredSaltLength">Represents the desired saltlength, if the Salt is longer or shorter it will change to the specified length</param>
+    /// <param name="PsuedoRandomNumber">Represents the PsuedoRandomNumber, which is used to transform Rfc2898DeriveByes with SHA1Managed into a ByteArray</param>
+
+    procedure HashRfc2898DeriveBytes(Password: Text; Salt: Text; DesiredSaltLength: Integer; PsuedoRandomNumber: Integer) Hash: Text
```

Comment:
After discussing with my colleagues, it does not make sense to have DesiredSaltLength as part of the API. We should not be modifying the salt in this function. 

The PadStr(Salt...) below should be carried out before calling this function.

Reply:
Oke, I added the PadStr in because if the Salt is too short it will give an error. It's ofcourse fine for us to check it beforehand. I will remove it.

Reply:
Having `DesiredSaltLength` doesn't really prevent an error from occurring. 

You need to handle the `ArgumentException` in a way that it does not pops out.